src/test/java/testsuite/integration/container/AuroraMySqlIntegrationTest.java

```diff
@@ -0,0 +1,596 @@
+    try (XSSFWorkbook workbook = new XSSFWorkbook()) {
+      // Title
+      final XSSFSheet sheet = workbook.createSheet("FailureDetectionResults");
+
+      // Header
+      Row row = sheet.createRow(0);
+      Cell cell;
+      final String[] headers = {"FailureDetectionGraceTime", "FailureDetectionInterval", "FailureDetectionCount", "SleepDelayMS", "MinFailureDetectionTime", "MaxFailureDetectionTime", "AvgFailureDetectionTime"};
+      for (int i = 0; i < headers.length; i++) {
+        cell = row.createCell(i);
+        cell.setCellValue(headers[i]);
+      }
+
+      // Data
+      for (int rows = 0; rows < dataList.size(); rows++) {
+        List<Integer> dataPoints = dataList.get(rows);
+        row = sheet.createRow(rows + 1);
+        for (int data = 0; data < dataPoints.size(); data++) {
+          cell = row.createCell(data);
+          cell.setCellValue(dataPoints.get(data));
+        }
+      }
+
+      // Write to file
+      final File newExcelFile = new File("./build/reports/tests/FailureDetectionResults.xlsx");
+      newExcelFile.createNewFile();
+      try (FileOutputStream fileOut = new FileOutputStream(newExcelFile)) {
+        workbook.write(fileOut);
+      }
+    }
+  }
+
+  @BeforeEach
+  public void setUpEach() {
+    proxyMap.forEach((k, v) -> {
+      try {
+        v.toxics().get("DOWN-STREAM").remove();
+        v.toxics().get("UP-STREAM").remove();
+      } catch (Exception e) {
+        // Ignore as toxics were not set
+      }
+    });
+  }
+
+  @ParameterizedTest(name = "test_ConnectionString")
+  @MethodSource("generateConnectionString")
+  public void test_ConnectionString(String connStr, int port) throws SQLException {
+    final Connection conn = connectToInstance(connStr, port);
+    assertTrue(conn.isValid(5));
+    conn.close();
+  }
+
+  private static Stream<Arguments> generateConnectionString() {
+    return Stream.of(
+        Arguments.of(MYSQL_INSTANCE_1_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT),
+        Arguments.of(MYSQL_CLUSTER_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT),
+        Arguments.of(MYSQL_RO_CLUSTER_URL, MYSQL_PORT),
+        Arguments.of(MYSQL_RO_CLUSTER_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT)
+    );
+  }
+
+  @Test
+  public void test_ValidateConnectionWhenNetworkDown() throws SQLException, IOException {
+    final Connection conn = connectToInstance(MYSQL_INSTANCE_1_URL + PROXIED_DOMAIN_NAME_SUFFIX, MYSQL_PROXY_PORT);
+    assertTrue(conn.isValid(5));
+
+    proxyInstance_1.toxics().bandwidth("DOWN-STREAM", ToxicDirection.DOWNSTREAM, 0); // from mysql server towards mysql driver
+    proxyInstance_1.toxics().bandwidth("UP-STREAM", ToxicDirection.UPSTREAM, 0); // from mysql driver towards mysql server
+
+    assertFalse(conn.isValid(5));
+
+    proxyInstance_1.toxics().get("DOWN-STREAM").remove();
```

Comment:
Probably it makes sense to extract these two line into a helper method with a good name. There's a lot of repetition of them.
enableConnectivity() ?